ui/pages/confirmations/components/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js

```diff
@@ -61,6 +68,26 @@ const ConfirmPageContainerNavigation = () => {
     }
   };
 
+  const pendingConfirmations = useSelector(pendingConfirmationsSortedSelector);
+  const queuedRequestCount = useSelector(getQueuedRequestCount);
+  const trackEvent = useContext(MetaMetricsContext);
+  const [metricsSent, setMetricsSent] = useState(false);
+
+  if (queuedRequestCount > 0 && !metricsSent) {
+    trackEvent({
```

Comment:
Side comment: in case this also enables the metrics being properties of the existing transactions and signatures events instead of separate events it would be amazing too. If not, we're good to continue as separate events.

Reply:
> Side comment: in case this also enables the metrics being properties of the existing transactions and signatures events instead of separate events it would be amazing too. If not, we're good to continue as separate events.

This is currently not possible, because the metric still needs to be sent from the UI code and only in the middlewares could we enrich existing events with `ui_customization` property as initially proposed.

As such, we are using the dedicated event.